app/models/site_status.rb

```diff
@@ -14,6 +14,12 @@
 class SiteStatus < ApplicationRecord
   self.table_name = "course_site"
 
+  validate if: Proc.new { |s| s.course&.study_mode.present? } do
+    unless vac_status_consistent_with_course_study_mode?
+      errors.add(:vac_status, "(#{vac_status}) must be consistent with course study mode #{course.study_mode}")
+    end
+  end
```

Comment:
I was trying to say earlier before my daughter a bit rudely interrupted 🤢 ... it'd be nice to move the body of this to a method. I know it already uses `vac_status_consistent_with_course_study_mode?`, which is good, but I think it be cleaner to keep this part of the model free of code (esp multi-line block) and just have declarations here.

Worth considering if you're going to move this to the other PR.

Reply:
Done.